### PR TITLE
[Cypress] coded regions, replace selector with region

### PR DESCRIPTION
**Pull request by SophieDePaula**
[Trello card](https://trello.com/c/uZxC9grz/1615-autodesk-incorrect-region-placement)
This PR purpose is to simplify the experience around coded regions. Currently, we support sending`selectore` and `region` properties, however, we support other options like `padding` and `regionId` only with `region` which is confusing for the customer.

TODO:
- [x] look at accessibility and floating regions - this is currently a limitation for elements, and it works when sending a selector.
- [x] Make sure we have tests to cover the changes and the added scenarios

### CODE REVIEW

File: js/packages/driver/src/element.ts [hunk 1]
```diff
@@ -59,6 +59,8 @@ export class Element<TDriver, TContext, TElement, TSelector> {
 
     if (specUtils.isSimpleCommonSelector(this._selector)) {
       this._commonSelector = this._selector
+    } else if (typeof this._selector === 'string') {
```
Comment (kyryloonufriiev) [inline js/packages/driver/src/element.ts:62]: use `utils.types.isString(...)`

**Review by kyryloonufriiev — CHANGES_REQUESTED** [review]

File: js/packages/eyes-cypress/test/coverage/generic/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "node": ">=8.0.0"
   },
   "devDependencies": {
-    "cypress": "latest",
+    "cypress": "^10.0.0",
```
Comment (kyryloonufriiev) [inline js/packages/eyes-cypress/test/coverage/generic/package.json:15]: You can return the `latest` here after our solution for cypress@11 right?

Comment (SophieDePaula) [reply js/packages/eyes-cypress/test/coverage/generic/package.json:15]: @kyryloonufriiev I noticed that I get a webpack error when I try to run our tests with Cypress 11 that I need to investigate and I didn't want to delay the release of this and the other PR I am working on.
I'll upgrade our tests to run with cypress 11 after this is done.

File: pending-changes.yaml [hunk 3]
```diff
@@ -91,7 +91,7 @@ Apollo SDK:
   feature:
     - Added new selector extensions `child` and `fallback`
   bug-fix:
-    - 
+    - Add support for regionId and padding for accessibility and floating regions
```
Comment (SophieDePaula) [inline pending-changes.yaml:94]: @benbabayoff8

File: js/packages/eyes-cypress/test/coverage/generic/package.json (hunk 2 above)
Comment (SophieDePaula) [reply js/packages/eyes-cypress/test/coverage/generic/package.json]: @kyryloonufriiev it seems that the webpack error no longer occurs with cypress 11.1.0

**Review by kyryloonufriiev — APPROVED** [review]
Looks good to me

**Pull request merged**